packages/astro/src/compiler/transform/postcss-scoped-styles/index.ts

```diff
@@ -13,6 +13,11 @@ interface Selector {
 const CSS_SEPARATORS = new Set([' ', ',', '+', '>', '~']);
 const KEYFRAME_PERCENT = /\d+\.?\d*%/;
 
+/** minify selector CSS */
+function minifySelector(selector: string): string {
+  return selector.replace(/(\r?\n|\s)+/g, ' ').replace(/\s*(,|\+|>|~|\(|\))\s*/g, '$1');
```

Comment:
The change here is CSS sanitization. This removes any extraneous whitespace characters (including linebreaks) that aren’t essential to the CSS selector (keeping in mind that _some_ whitespace is essential as child selector, but other whitespace is non-essential such as `.foo   +   \n .bar` and can be removed).

The issue was whitespace not being handled properly, resulting in things like ` body` (rather than `body`) being a selector and incorrectly marked as needing scoping.